packages/grid-pro/test/edit-column-renderer.test.js

```diff
@@ -251,6 +251,25 @@ describe('edit column renderer', () => {
 
       expect(getContainerCellContent(grid.$.items, 0, 0).innerHTML).to.equal('Bar');
     });
+
+    it('should not exit the edit mode if clear button is clicked in the text-field', async () => {
+      column.editModeRenderer = function (root, _, model) {
+        root.innerHTML = '';
+        const field = document.createElement('vaadin-text-field');
+        field.value = model.item.name;
+        field.clearButtonVisible = true;
+        root.appendChild(field);
+      };
+      dblclick(cell._content);
+      editor = getCellEditor(cell);
+
+      // Simulate a mouse click on the clear button wich first focus out from the field
+      focusout(editor);
+      // and then the click is done
+      mousedown(editor.$.clearButton);
```

Comment:
This test is a bit artificial because in fact these events are fired in the reverse sequence: `mousedown` triggers `focusout` unless `event.preventDefault()` is called - e.g. how we do it in `vaadin-combo-box`:

https://github.com/vaadin/web-components/blob/7cc4252e925d318d49aacf00b2e8878f1641299e/packages/combo-box/src/vaadin-combo-box-mixin.js#L579-L582

Reply:
I first tried make the test use `click` instead and that happens after `focusout`, but then switched to `mousedown` which indeed happens before `focusout` but didn't think about refactoring the test.

Reply:
I am actually not sure how I can simulate the same behavior (`mousedown` followed by `focusout`) on the test so that it fails with the current state and passes after the changes. Do you have any idea how it can be done?

Reply:
I think the only reliable way to test this would be to use `sendMouse` plugin.